blockchain/contracts/DiceLibrary.sol

```diff
@@ -4,54 +4,97 @@ pragma solidity ^0.8.0;
 import "hardhat/console.sol";
 
 library DiceLibrary {
-
     struct Dice {
         string name;
         uint8 sides;
+        // TODO: Store colors as bytes3, and translate to hex chars
+        string fgColor;
+        string bgColor;
+        uint8 font;
         bool zeroBased;
-        Geometries geometry;
     }
 
     struct DiceStorage {
         mapping(uint256 => Dice) dice;
     }
 
-    enum Geometries {
-        Standard,
-        Rhombic,
-        Elongated
-    }
-
-    function createDice(DiceStorage storage self, string memory name, uint256 tokenId, uint8 sides) internal {
+    function createDice(
+        DiceStorage storage self,
+        uint256 tokenId,
+        string calldata name,
+        uint8 sides,
+        string calldata fgColor,
+        string calldata bgColor,
+        uint8 font
+    ) internal {
         require(sides > 0, "Sides must be non-zero");
 
-        // TODO: Have these settable by the caller
-        Geometries geometry = Geometries.Standard;
-        // TODO: Do any dice besides D10 have "0" faces?
-        bool zeroBased = (sides == 10) ? true : false;
-        self.dice[tokenId] = Dice(name, sides, zeroBased, geometry);
+        bool zeroBased = (sides == 10) ? true : false; // Only D10s have "0"
+        self.dice[tokenId] = Dice(
+            name,
+            sides,
+            fgColor,
+            bgColor,
+            font,
+            zeroBased
+        );
     }
 
     // URI path is composed of attributes
-    function getTokenURIpath(DiceStorage storage self, uint256 tokenId) internal view returns (string memory) {
+    function getTokenURIpath(DiceStorage storage self, uint256 tokenId)
+        internal
+        view
+        returns (string memory)
+    {
         // TODO: add other dice attributes in later commit
         // return(string(abi.encodePacked("/", self.dice[tokenId].sides,"/", self.dice[tokenId].sides)));
         return "/sides/color/font";
     }
 
-    function getTraits(DiceStorage storage self, uint256 tokenId) public view 
-        returns (string memory name, string memory tokenURIpath, uint8 sides) {
-        return (self.dice[tokenId].name, getTokenURIpath(self, tokenId), self.dice[tokenId].sides);
+    function getTraits(DiceStorage storage self, uint256 tokenId)
+        public
+        view
+        returns (
+            string memory name,
+            uint8 sides,
+            string memory fgColor,
+            string memory bgColor,
+            uint8 font
+        )
+    {
+        return (
+            self.dice[tokenId].name,
+            self.dice[tokenId].sides,
+            self.dice[tokenId].fgColor,
```

Comment:
In this case, we're saving the colors for every NFT. I had expected a lookup table instead. So instead of saving the whole color every time, we would save a single index that, when sent to a method, would return the appropriate color. That should store considerably less data.